### PR TITLE
Fix region issues on Azure MySQL

### PR DESCRIPTION
**Pull request by shenqianjin**
Default region:
image

List same with portal:
image

Unavailable location alert:
image

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java [hunk 1]
```diff
@@ -5,31 +5,83 @@
...
+    @org.jetbrains.annotations.NotNull
+    @Override
+    public AzureValidationInfo doValidate() {
+        final AzureValidationInfo info = super.doValidate();
+        if (!AzureValidationInfo.OK.equals(info)) {
+            return info;
+        }
+        MySQLManager manager = AuthMethodManager.getInstance().getMySQLManager(subscription.subscriptionId());
+        List<PerformanceTierPropertiesInner> propertiesInnerList = manager.locationBasedPerformanceTiers().inner().list(getValue().name());
+        if (CollectionUtils.size(propertiesInnerList) < 3) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java:81]: `CollectionUtils.size(propertiesInnerList) < 3` refers the region is illegal? Could you please add some comment here?

File: Utils/pom.xml [hunk 2]
```diff
@@ -45,7 +45,8 @@
         <azure.identity.version>1.2.1</azure.identity.version>
         <azure.resourcemanager.version>2.1.0</azure.resourcemanager.version>
         <azure.toolkit-auth.version>${project.version}</azure.toolkit-auth.version>
-        <azure.version>1.38.0</azure.version>
+        <azure.tools-common.version>0.10.2</azure.tools-common.version>
```
Comment (Flanker32) [inline Utils/pom.xml:48]: We have removed the dependency for tools-common, why add it back here?

Comment (shenqianjin) [reply Utils/pom.xml]: removed

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java (hunk 1 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java]: added

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java [hunk 3]
```diff
@@ -5,31 +5,85 @@
...
+    @Override
+    public AzureValidationInfo doValidate() {
+        final AzureValidationInfo info = super.doValidate();
+        if (!AzureValidationInfo.OK.equals(info)) {
+            return info;
+        }
+        MySQLManager manager = AuthMethodManager.getInstance().getMySQLManager(subscription.subscriptionId());
+        List<PerformanceTierPropertiesInner> propertiesInnerList = manager.locationBasedPerformanceTiers().inner().list(getValue().name());
+        // 'Basic' performance tier represents the service is available in this location for your subscription
+        if (propertiesInnerList.stream().filter(e -> DEFAULT_MYSQL_PERFORMANCE_TIER_ID.equals(e.id())).count() > 0L) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java:83]: May use `Stream.anyMatch()` here

**Review by Flanker32 — DISMISSED** [review]
LGTM~

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java (hunk 3 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java]: good idea.

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java [hunk 4]
```diff
@@ -122,6 +125,16 @@ public static void stop(final String subscriptionId, final Server server) {
...
     }
 
+    public static List<Region> listSupportedRegions(Subscription subscription) {
+        PagedList<Location> locationList = subscription.listLocations();
+        locationList.loadAll();
+        return locationList.stream()
+                .filter(e -> RegionType.PHYSICAL.equals(e.regionType()))
+                .filter(e -> MYSQL_SUPPORTED_REGIONS.contains(e.name()))
+                .map(e -> e.region())
+                .collect(Collectors.toList());
```
Comment (andxu) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java:135]: distinct?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java [hunk 5]
```diff
@@ -5,31 +5,85 @@
...
+        if (Objects.isNull(subscription)) {
+            return new ArrayList<>();
+        }
+        return MySQLMvpModel.listSupportedRegions(subscription);
     }
 
     @Override
     protected String getItemText(Object item) {
         if (item instanceof Region) {
+            if ("centraluseuap".equals(((Region) item).label())) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/MySQLRegionComboBox.java:58]: TODO: will be deleted

**Review by andxu — APPROVED** [review]

**Pull request merged**